sdk/core/amqp-common/src/auth/aad.ts

```diff
@@ -45,18 +45,9 @@ export class AadTokenProvider implements TokenProvider {
       | DeviceTokenCredentials
       | MSITokenCredentials
   ) {
-    if (
-      !credentials ||
-      (credentials &&
-        !(
-          credentials instanceof ApplicationTokenCredentials ||
-          credentials instanceof UserTokenCredentials ||
-          credentials instanceof DeviceTokenCredentials ||
-          credentials instanceof MSITokenCredentials
-        ))
-    ) {
+    if (!credentials || !credentials.getToken) {
       throw new TypeError(
-        "'credentials' is a required parameter and must be an instance of " +
+        "'credentials' is a required parameter and must implement getToken() such as in " +
```

Comment:
We usually want to avoid making changes to the error message when we are making non breaking changes.
There might be customers who are parsing the error message and then taking subsequent actions in their application. Changing the text on error message can break them.

Reply:
Reverted. Ideally though, I'm in favor of have error code and description model, since strings and language can change all the time